src/components/Dialog/Dialog.js

```diff
@@ -26,19 +26,32 @@ function Dialog(props) {
   const dispatch = useDispatch();
   const { body, isModal } = props;
 
-  const escFunction = () => {
-    dispatch(setDialogVisible());
-  };
+  const dialogRef = useRef(null);
+  let activeElement;
 
-  const registerEsc = isRegister => {
-    if (isRegister) {
-      document.addEventListener('keydown', escFunction, false);
-    } else {
-      document.removeEventListener('keydown', escFunction, false);
-    }
-  };
+  const escFunction = e => e.which === 27 && dispatch(setDialogVisible());
 
   useEffect(() => {
+    const registerEsc = isRegister => {
+      if (isRegister) {
+        document.addEventListener('keydown', escFunction, false);
+
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+        activeElement = document.activeElement;
+        const dialogFocusables = dialogRef.current.ref.current.querySelectorAll(
+          '[tabindex="0"]',
+        );
+
+        if (dialogFocusables.length) {
+          // Focus the first focusable element in the popup.
+          dialogFocusables[0].focus();
+        }
+      } else {
+        document.removeEventListener('keydown', escFunction, false);
+        // Re focus the element that opened the dialog.
+        activeElement.focus();
+      }
+    };
```

Comment:
I meant the content of the function not the function itself.
you don't need to encapsulate it into a function

Reply:
I see what you mean now sorry :)